miracl/seg/ace_deploy_model.py

```diff
@@ -639,8 +640,11 @@ def deploy_functions(
     gpu_opt = cfg["general"].get("GPU", "single")
     global device
     if gpu_opt == "single":
-        device = torch.device("cuda:0")
+        if torch.cuda.device_count() <= gpu_index:
```

Comment:
I don't think we have this check in MIRACL overall. I agree we should have something that checks and will add it to the docker build so that if the user has a GPU, docker will be set up correctly. And in the case they don't have a GPU then some functionality won't be available (like ACE for example)

Reply:
I'll have a look at this today. Maybe sth in the build script rather than in ACE directly...